e-digi/src/br/com/caelum/editora/modelos/Livro.java

```diff
@@ -0,0 +1,122 @@
+package br.com.caelum.editora.modelos;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+public class Livro {
+	private static final int TAMANHO_MINIMO_DO_RESUMO = 500;
+	
+	private String titulo;
+	private String resumo;
+	private String sumario;
+	private int paginas;
+	private String isbn;
+	private Autor autor;
+	private Categoria categoria;
+	private int edicao;
+	private BigDecimal preco;
+	private Instant criadoEm;
+	
+	public Livro() {
+		this.criadoEm = Instant.now();
+	}
```

Comment:
Aqui você tá permitindo que o livro seja criado em estado inválido, não existe obrigatoriedade nenhuma na passagem dos parâmetros.

Reply:
Mas nesse caso, passar todos os parâmetros também não iria ficar um construtor muito grande? Ou deveria passar pelos menos um parâmetro ?

Reply:
Qual é o problema do tamanho do construtor?

Reply:
Acredito que ficaria um código meio difícil de se ler, pelo menos se todos os atributos estivessem dentro do construtor. Ficaria algo como:

`public Livro(String titulo, String resumo, String sumario, int paginas, String isbn, Autor autor,
			Categoria categoria, int edicao, BigDecimal preco) {
		super();
		this.setTitulo(titulo);
		this.setResumo(resumo);
		this.setSumario(sumario);
		this.setPaginas(paginas);
		this.setIsbn(isbn);
		this.setAutor(autor);
		this.setCategoria(categoria);
		this.setEdicao(edicao);
		this.setPreco(preco);
	}`
Visualmente eu acho que ficaria estranho,principalmente quando levamos em consideração que o resumo precisa ter pelo menos 500 caracteres. Porém se não há problema em te-lo desse tamanho tudo bem, afinal da maneira como ele está agora realmente permite que ele seja criado em estado inválido.

Reply:
Uma das diretrizes diz que não podemos permitir objetos criados em estado inválido, correto?

Reply:
Certo, entendi, mesmo que o construtor fique grande, um objeto precisa ser criado com um estado válido. Vou colocar os parâmetros no construtor.